<div class="science-app" data-app-name="tree-viewer-app">
    <h3>Tree Viewer</h3>
    <div class="row">
        <form id="tree-render" class="col-sm-5">
            <div class="form-group form-inline">
                <label class="control-label" for="locus_id">Query for a gene tree that contains AGI ID (e.g. AT3G52430).</label>
                <input type="text" id="locus_id" name="locus_id" class="form-control" value="AT3G52430" placeholder="Gene name (AGI ID)"/>
                <button class="btn btn-primary" type="button" id="queryButton">Query</button>
            </div>
            <div class="form-group">
                <label class="control-label" for="tree">Or upload a Newick tree format file.</label>
                <input type="file" id="file" name="file"/>
            </div>
            <div class="form-group">
                <label class="control-label" for="tree">Or paste a Newick tree format text string here.</label>
                <textarea rows="3" class="form-control" name="tree" id="tree">(((C.elegans,Fly),(((((((((Tasmanian Devil,Wallaby,Opossum),((Armadillo,Sloth),(Rock hyrax,Tenrec,Elephant),(((Rabbit,Pika),(((Rat,Mouse),Kangaroo rat,Squirrel),Guinea Pig)),((Mouse lemur,Bushbaby),((((((Chimp,Human,Gorilla),Orangutan),Gibbon),Macaque),Marmoset),Tarsier)),Tree Shrew),((Microbat,Flying fox),(Hedgehog,Shrew),((Panda,Dog,Domestic ferret),Cat),((Cow,Sheep),Pig,Alpaca,Dolphin),Horse))),Platypus),((((Collared flycatcher,Zebra finch),(Chicken,Turkey),Duck),Chinese softshell turtle),Anole lizard)),Xenopus),Coelacanth),(((Zebrafish,Cave fish),((((Medaka,Platyfish),Stickleback),(Fugu,Tetraodon),Tilapia),Cod)),Spotted gar)),Lamprey),(C.savignyi,C.intestinalis))),S.cerevisiae);</textarea>
            </div>
            <div class="form-group">
                <button class="btn btn-success" type="submit" id="drawButton">Draw</button>
                <button class="btn btn-danger" type="button" id="clearButton">Clear</button>
                <button class="btn btn-info btn-clipboard" type="button" id="copyButton" data-clipboard-target="#tree"><i class="fa fa-clipboard"></i> Copy to Clipboard</button>
            </div>
        </form>
    </div>

    <br>

    <div id="error"></div>
    <div id="progress_region" class="hidden">
        <h4>Searching...</h4>
        <div class="progress progress-striped active">
            <div class="progress-bar" role="progressbar" aria-valuenow="100" aria-valuemin="0" aria-valuemax="100" style="width: 100%;">
                <span class="sr-only">Loading Data</span>
            </div>
        </div>
    </div>

    <br>

    <!-- Navigation tabs -->
    <ul class="nav nav-tabs" role="tablist">
        <li id="about-tab" class="active"><a href="#about" aria-controls="about" role="tab" data-toggle="tab">About</a></li>
        <li id="tree-tab"><a href="#tree-space" aria-controls="tree-space" role="tab" data-toggle="tab">Tree</a></li>
    </ul>

    <!-- Tab panes -->
    <div class="tab-content">
        <div role="tabpanel" class="tab-pane fade in active" id="about">
            <a href="http://ensemblgenomes.org/" target="_blank"><img src="http://ensemblgenomes.org/sites/ensemblgenomes.org/themes/ensemblgenomes/e-ensemblgenomes.png" alt="Ensembl Genomes logo"/></a>
            <br>
            <p>This app draws trees from data in <a href="http://evolution.genetics.washington.edu/phylip/newicktree.html" target="_blank">Newick tree format <i class="fa fa-external-link"></i></a>.
            The visualization is done using a <a href="https://www.biojs.net/" target="_blank">BioJS <i class="fa fa-external-link"></i></a> module called <a href="http://tntvis.github.io/tnt.tree/" target="_blank">TnT Tree <i class="fa fa-external-link"></i></a>
            that is based on the JavaScript library for manipulating data called <a href="https://d3js.org/" target="_blank">D3 <i class="fa fa-external-link"></i></a>.
            The main data source is EMBL-EBI's <a href="http://ensemblgenomes.org/" target="_blank">Ensembl Genomes <i class="fa fa-external-link"></i></a> Gene tree. Ensembl gene trees are generated by the <a href="http://useast.ensembl.org/info/genome/compara/homology_method.html" target="_blank">Gene Orthology/Paralogy prediction method pipeline <i class="fa fa-external-link"></i></a>.
            All homologues in Ensembl are determined from gene trees. Gene trees are constructed using one representative protein for every gene in every species in Ensembl.  The longest translation annotated by the <a href="http://useast.ensembl.org/info/genome/genebuild/ccds.html" target="_blank">CCDS <i class="fa fa-external-link"></i></a>
            project is used, if any are available, or the longest protein-coding translation otherwise.  (The trees can also be considered as protein trees).</p>
            <h5>EMBL-EBI asks that the following paper be cited when using Ensembl Genomes data:</h5>
                <ul>
                    <li>Kersey PJ, Allen JE, Armean I, Boddu S, Bolt BJ, Carvalho-Silva D, Christensen M, Davis P, Falin LJ,
                        Grabmueller C, Humphrey J, Kerhornou A, Khobova J, Aranganathan NK, Langridge N, Lowy E, McDowall MD,
                        Maheswari U, Nuhn M, Ong CK, Overduin B, Paulini M, Pedro H, Perry E, Spudich G, Tapanari E, Walts B,
                        Williams G, Tello-Ruiz M, Stein J, Wei S, Ware D, Bolser DM, Howe KL, Kulesha E, Lawson D, Maslen G,
                        Staines DM. <strong>Ensembl Genomes 2016: more genomes, more complexity</strong>. <i>Nucleic Acids Res. 2016 Jan 4;44(D1):D574-80.</i> (<a href="http://www.ncbi.nlm.nih.gov/pubmed/26578574" target="_blank">PMID:26578574 <i class="fa fa-external-link"></i></a>)</li>
                </ul>

            <h4>Instructions</h4>
            <p>To get started, select from one of the following methods for inputting a tree:
                <ul>
                    <li>Enter an AGI locus identifier and click the <strong>Query</strong> button to grab a Gene tree from Ensembl Genomes.</li>
                    <li>Click on the <strong>Choose File</strong> button and pick a file containing a tree in Newick format.</li>
                    <li>Paste text in Newick tree format into the text box.</li>
                </ul>
                And then click on the <strong>Draw</strong> button. The tree will be drawn in the <strong>Tree</strong> tab.
            </p>

            <h5>Features (Tree)</h5>
            <p>
                <ul>
                    <li>Change the layout of the tree between vertical and radial using the <strong>Layout</strong> menu.</li>
                    <li>Highlight nodes in the tree (in red) by typing in the <strong>Search in tree</strong> text box.</li>
                    <li>Nodes in the tree can be collapsed/expanded by clicking on a node.</li>
                    <li>When a tree is queried or uploaded from a file, the Newick tree format text is pasted into the text box and can be saved by clicking the <strong>Copy to Clipboard</strong> button.</li>
                </ul>
            </p>

            <h4>Authors</h4>
            <p>
                <ul>
                    <li>Erik Ferlanti, JCVI</li>
                </ul>
            </p>
        </div>

        <div role="tabpanel" class="tab-pane fade" id="tree-space">
            <div id="tree-canvas" class="results-panel"><h4>Please select data to draw a tree.</h4></div>
        </div>
    </div>

    <br>

    <div class="row">
        <div class="col-sm-12">
            <p>
                Data &amp; web services provided by
                <a href="http://ensemblgenomes.org/" target="_blank">EnsemblGenomes <i class="fa fa-external-link"></i></a>.
            </p>
        </div>
    </div>
</div>
